src/d3a/models/strategy/__init__.py

```diff
@@ -56,78 +56,89 @@ class Offers:
 
     def __init__(self, strategy):
         self.strategy = strategy
-        self.bought = {}  # type: Dict[Offer, Market]
-        self.posted = {}  # type: Dict[Offer, Market]
-        self.sold = {}  # type: Dict[Market, List[str]]
+        self.bought = {}  # type: Dict[Offer, Str]
+        self.posted = {}  # type: Dict[Offer, Str]
+        self.sold = {}  # type: Dict[Str, List[str]]
         self.changed = {}  # type: Dict[str, Offer]
 
+    @property
+    def area(self):
+        # TODO: Remove the owner and area distinction from the AreaBehaviorBase class
+        return self.strategy.area if self.strategy.area is not None else self.strategy.owner
+
+    def filter_offers_in_future_markets(self, existing_offers):
```

Comment:
AFAIU, this function deletes all "past" offers right? Very minor, but if so, can you please rename the function accordingly, maybe to "delete_past_offers" or so? Thanks!

Reply:
delete_past_offers sounds great. I will also rename accordingly the delete_past_markets method to delete_past_markets_offers .